lib/dry/schema/macros/hash.rb

```diff
@@ -12,7 +12,34 @@ class Hash < Schema
         # @api private
         def call(*args, &block)
           trace << hash?
-          super(*args, &block)
+
+          super(*args) unless args.empty?
+
+          if block
+            definition = schema_dsl.new(&block)
+
+            parent_type = schema_dsl.types[name]
+            definition_schema = definition.type_schema
+
+            schema_type =
+              if parent_type.respond_to?(:of)
+                parent_type.of(definition_schema)
+              else
+                definition_schema
+              end
+
+            final_type =
+              if parent_type.optional?
+                schema_type.optional
+              else
+                schema_type
+              end
+
+            schema_dsl.set_type(name, final_type)
+
+            trace << definition.to_rule
+          end
+
```

Comment:
This looks like a mistake in conflict resolution, as `Macros::Schema` should include this whole thing whereas `Macros::Hash` simply [prepends `hash?` check and calls super](https://github.com/dry-rb/dry-schema/blob/master/lib/dry/schema/macros/hash.rb#L14-L16).

Specs are passing because the code does the same thing, it's just that it's duplicated now in this branch.

Reply:
totally! fixed